### PR TITLE
Feature: Remove parentheses in price difference column.

### PR DESCRIPTION
**Pull request by Just-Existing**
## Feature

This PR addresses some of the problems described in issue #9203

## Summary
I removed the parentheses from the price difference column.

## Test Screenshots
Before: 
price_difference_column_test_before
After:
price_difference_column_test_after
show parenthesis

## Checklist
To fulfill the consensus on this pull request, these criteria must be met:

- [x] Remove parentheses.
- [x] Add option to put parens back in.

(I moved the original checklist to [this gist](https://gist.github.com/Just-Existing/d3651c92cc446dcd293ed045a534fe9a))

### CODE REVIEW

**Comment by OcelotWalrus** [comment]
Why would you remove the parentheses?

**Comment by Zireael07** [comment]
@OcelotWalrus Read the linked issue, it explains why

**Comment by OcelotWalrus** [comment]
Woops, sorry.

**Comment by mOctave** [comment]
Honestly, I'm not a huge fan of removing the parentheses. I guess I should have mentioned this on the issue last year, but I seem to have missed it.

For me, the parentheses help to distinguish between what's a trade price and what isn't in a way that the +/- doesn't. It also helps pad one-digit differences so that they have a more similar width to everything else and don't stand out quite so much. With the image below, my eye is immediately drawn to metal, which is the most useless trade price on the entire board.
Screenshot 2024-03-07 at 9 42 19 PM

**Comment by Hecter94** [comment]
The linked issue suggested this as a potential replacement for the parentheses if further visual separation was needed
> I also recommend dropping the parentheses because they don't serve any function and take up precious space. The numbers are demarcated from the names by being right-aligned. If further visual separation is needed, I suggest drawing a vertical grid line between the 2 columns (rather than the unintentional squiggly line formed by the left parentheses).

Thoughts?

**Comment by Zireael07** [comment]
The grid is a good idea

**Comment by Amazinite** [comment]
Keep in mind that #9721 is open as well.

**Comment by Zireael07** [comment]
In the words of a famous meme, why not both? They're orthogonal solutions AND map shades shouldn't be the only way to do it because of color-blindness

**Comment by pepoluan** [comment]
> For me, the parentheses help to distinguish between what's a trade price and what isn't in a way that the +/- doesn't. It also helps pad one-digit differences so that they have a more similar width to everything else and don't stand out quite so much. With the image below, my eye is immediately drawn to metal, which is the most useless trade price on the entire board.

Not everyone has the same reaction.

For me, personally, removing the parentheses exposes the number better.

Maybe because I'm so familiar with Excel spreadsheets nearly every workday, but still we now have 2 anecdotal evidence that people have different preferences.

Maybe put in as an option? So those that prefer the parens can keep them, while those that prefer naked numbers can remove the parens.

> Remove parentheses.
> Replace hyphen(-) with minus(−).
> Make font monospace.

I would say from top to bottom represents a diminishing return. Even just removing the parens already improves the UI signficantly for me, as my eyes can more easily see the leading character.

If it's impractical to replace all in one go, we can do it in steps. Starting with the (possibly optional) removal of the parens.

**Comment by mOctave** [comment]
> Maybe put in as an option? So those that prefer the parens can keep them, while those that prefer naked numbers can remove the parens.

I'd support a setting.

File: source/MapDetailPanel.cpp [hunk 1]
```diff
@@ -800,11 +800,16 @@ void MapDetailPanel::DrawInfo()
 			else
 			{
 				value -= localValue;
-				price += "(";
+				if(Preferences::Has(Preferences::PRICE_PARENS))
```
Comment (TomGoodIdea) [inline source/MapDetailPanel.cpp:803]: Other preference checks just use the preference name directly. If you want to change it, open a separate refactor PR for all preferences.
```suggestion
				if(Preferences::Has("Show parenthesis on Job Board"))
```

File: source/MapDetailPanel.cpp [hunk 2]
```diff
@@ -800,11 +800,16 @@ void MapDetailPanel::DrawInfo()
...
+					price += "(";
 				if(value > 0)
 					price += '+';
+				else if(value < 0)
+					price += '-';
+				value = abs(value);
 				price += to_string(value);
-				price += ")";
+				if(Preferences::Has(Preferences::PRICE_PARENS))
```
Comment (TomGoodIdea) [inline source/MapDetailPanel.cpp:811]: (see above)
```suggestion
				if(Preferences::Has("Show parenthesis on Job Board"))
```

File: source/Preferences.h [hunk 3]
```diff
@@ -103,6 +103,8 @@ class Preferences {
 		BOTH
 	};
 
+	inline static const std::string PRICE_PARENS = "Show parenthesis on Job Board";
+
```
Comment (TomGoodIdea) [inline source/Preferences.h:107]: (see above)
```suggestion
```

File: source/PreferencesPanel.cpp [hunk 4]
```diff
@@ -692,7 +692,8 @@ void PreferencesPanel::DrawSettings()
 		"Interrupt fast-forward",
 		"Landing zoom",
 		SCROLL_SPEED,
-		DATE_FORMAT
+		DATE_FORMAT,
+		Preferences::PRICE_PARENS
```
Comment (TomGoodIdea) [inline source/PreferencesPanel.cpp:696]: (see above)
```suggestion
		"Show parenthesis on Job Board"
```

File: source/MapDetailPanel.cpp [hunk 5]
```diff
@@ -800,11 +800,16 @@ void MapDetailPanel::DrawInfo()
...
 			{
 				value -= localValue;
-				price += "(";
+				if(Preferences::Has(Preferences::PRICE_PARENS))
+					price += "(";
 				if(value > 0)
 					price += '+';
+				else if(value < 0)
+					price += '-';
+				value = abs(value);
```
Comment (TomGoodIdea) [inline source/MapDetailPanel.cpp:809]: Wouldn't be the same without this code?

File: source/Preferences.cpp [hunk 6]
```diff
@@ -36,6 +36,8 @@ namespace {
 	map<string, bool> settings;
 	int scrollSpeed = 60;
 
+	bool PriceParens = false;
+
```
Comment (TomGoodIdea) [inline source/Preferences.cpp:40]: unused
```suggestion
```

File: source/Preferences.cpp [hunk 7]
```diff
@@ -730,3 +733,8 @@ int Preferences::GetPreviousSaveCount()
 {
 	return previousSaveCount;
 }
+
+bool Preferences::GetPriceParens()
+{
+	return PriceParens;
+}
```
Comment (TomGoodIdea) [inline source/Preferences.cpp:740]: unused
```suggestion
```

File: source/Preferences.h [hunk 8]
```diff
@@ -188,6 +190,8 @@ class Preferences {
 	static bool DoAlertHelper(AlertIndicator toDo);
 
 	static int GetPreviousSaveCount();
+
+	static bool GetPriceParens();
```
Comment (TomGoodIdea) [inline source/Preferences.h:194]: unused
```suggestion
```

File: source/Preferences.cpp [hunk 9]
```diff
@@ -153,6 +155,7 @@ void Preferences::Load()
 	// These settings should be on by default. There is no need to specify
 	// values for settings that are off by default.
 	settings["Landing zoom"] = true;
+	settings["Show parenthesis on Job Board"] = false;
```
Comment (TomGoodIdea) [inline source/Preferences.cpp:158]: no need to set a default value for preferences defaulting to "off"
```suggestion
```

File: source/MapDetailPanel.cpp (hunk 5 above)
Comment (Just-Existing) [reply source/MapDetailPanel.cpp:809]: At one point, I did not realize the difficulty of changing out the '-' character for a better fitting one.

File: source/Preferences.cpp (hunk 9 above)
Comment (jrincayc) [reply source/Preferences.cpp]: Hm, shouldn't this default to `true` to match the current default?

Comment (Just-Existing) [reply source/Preferences.cpp]: The lack of parenthesis is the more desirable behavior, see #9203.

File: source/PreferencesPanel.cpp [hunk 10]
```diff
@@ -692,7 +692,8 @@ void PreferencesPanel::DrawSettings()
 		"Interrupt fast-forward",
 		"Landing zoom",
 		SCROLL_SPEED,
-		DATE_FORMAT
+		DATE_FORMAT,
+		"Show parenthesis on Job Board"
```
Comment (TomGoodIdea) [inline source/PreferencesPanel.cpp:696]: Trade prices aren't shown on the Job Board screen, they are part of the "Ports" map mode.

Also please add a tooltip.

**Comment by TomGoodIdea** [comment]
@Just-Existing can you finish this?

**Comment by Just-Existing** [comment]
No. I have no motivation to learn how to create tooltips in this codebase.

**Comment by TomGoodIdea** [comment]
Just add an entry to the settings section in `data/_ui/tooltips.txt`. It's plain text, no coding required.

**Comment by Just-Existing** [comment]
> Just add an entry to the settings section in `data/_ui/tooltips.txt`.

Done. Thanks for the help.

**Review by TomGoodIdea — APPROVED** [review]

**Review by tibetiroka — APPROVED** [review]

**Comment by Zireael07** [comment]
yay!

**Pull request merged**